src/Microsoft.AspNet.Hosting.WindowsService/Program.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNet.Hosting.Internal;
+using Microsoft.Extensions.Configuration;
+using System.ServiceProcess;
+
+namespace Microsoft.AspNet.Hosting.WindowsService
+{
+    public class Program : ServiceBase
+    {
```

Comment:
The user can use this package as the entry point to his app, or he can derive from it. I see what you are saying here. I can pull the meat out the same way it was pulled out of Hosting.Program and placed into Hosting.WebApplication, but what should that class be called? WindowServiceWebApplication? It's still a web app.